demo/src/screens/componentScreens/IncubatorTextFieldScreen.tsx

```diff
@@ -0,0 +1,180 @@
+
+  resetFieldValue = () => {
+    this.input2.current?.clear();
+  }
+
+  render() {
+    const {errorPosition} = this.state;
+    return (
+      <ScrollView keyboardShouldPersistTaps="always">
+        <View flex padding-page>
+          <Text h1>TextField</Text>
+          <Text h3 blue50 marginV-s4>
+            Default
+          </Text>
+
+          <TextField
+            ref={this.input}
+            label="Name"
+            placeholder="Enter first and last name"
+          />
+
+          <Text h3 blue50 marginV-s4>
+            Static vs Floating Placeholder
+          </Text>
+          <View row bottom>
+            <TextField
+              placeholder="Floating placeholder"
+              floatingPlaceholder
+              floatingPlaceholderColor={{
+                focus: Colors.grey10,
+                default: Colors.grey30
+              }}
+              // floatingPlaceholderStyle={Typography.text60}
+              // style={Typography.text60}
+              containerStyle={{flex: 1}}
+              fieldStyle={styles.withUnderline}
+            />
+            <TextField
+              placeholder="Placeholder"
+              containerStyle={{flex: 1, marginLeft: Spacings.s4}}
+              fieldStyle={styles.withUnderline}
+            />
+          </View>
+          <Text h3 blue50 marginV-s4>
+            Leading/Trailing Button
+          </Text>
+
+          <TextField
+            ref={this.input2}
+            placeholder="Enter text..."
+            leadingButton={{iconSource: Assets.icons.demo.search}}
+            trailingButton={{iconSource: Assets.icons.demo.refresh, onPress: this.resetFieldValue}}
+            fieldStyle={styles.withUnderline}
+          />
+
+          <View row marginV-s4 spread>
+            <Text h3 blue50>
+              Validation
+            </Text>
+            <Button
+              size={Button.sizes.xSmall}
+              label={`Error Position: ${_.upperCase(errorPosition)}`}
+              onPress={() =>
+                this.setState({
+                  errorPosition:
+                    errorPosition === TextField.validationMessagePositions.TOP
+                      ? TextField.validationMessagePositions.BOTTOM
+                      : TextField.validationMessagePositions.TOP
+                })
+              }
+            />
+          </View>
+
+          <TextField
+            value={this.state.value}
+            onChangeText={(value) => this.setState({value})}
+            label="Email"
+            placeholder="Enter email"
+            enableErrors
+            validationMessage="Email is invalid"
+            validationMessageStyle={Typography.text90R}
+            validationMessagePosition={errorPosition}
+            validate={'email'}
+            validateOnChange
+            // validateOnStart
+            // validateOnBlur
+            fieldStyle={styles.withUnderline}
+          />
+
+          <Text h3 blue50 marginV-s4>
+            Colors By State
```

Comment:
This example is not working for me (testing iOS only for now)

Reply:
Fixed. There were couple of issues there. 
- I didn't implement the "error" state
- The default color was wrong

I also remove one of the example for "Color By State". 
One is good enough.